### PR TITLE
feat: OAuth2 database field

### PR DESCRIPTION
**Pull request by betodealmeida**
### SUMMARY

More work on Snowflake OAuth2 (this PR depends on #30082). This PR introduces a new field for the database configuration modal, allowing OAuth2 client information to be provided by the user:

Screenshot 2024-09-02 at 8 57 34 AM

The component combines the 5 different input fields, and then passes them inside the `masked_encrypted_extra` attribute on database CRUD:

```json
{
  "oauth2_client_info": {
    "id": "123456",
    "secret": "XXXXXXXXXXXX",
    "authorization_request_uri": "https://account.snowflakecomputing.com/oauth/authorize",
    "token_request_uri": "https://account.snowflakecomputing.com/oauth/token-request",
    "scope": "refresh_token session:role:USERADMIN"
  }
}
```

(Note that ideally we would pass these in the `parameters` field, and let the DB engine spec build the object in `encrypted_extra`. But the `build_sqlalchemy_uri` method — as the name suggests — only builds an URI, so we would have to modify it to return a tuple with `(uri, connect_args)` if we passed them client information inside `parameters`, and also figure out a way to inform that these fields should be stored in `encrypted_extra`, not `extra`, so for now the frontend does the work. I left a small TODO for me to fix this in the future.)

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

See above.

### TESTING INSTRUCTIONS

Adding tests.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/features/databases/DatabaseModal/DatabaseConnectionForm/CommonParameters.tsx [hunk 1]
```diff
@@ -161,23 +161,20 @@ export const httpPathField = ({
   getValidation,
   validationErrors,
   db,
-}: FieldPropTypes) => {
-  console.error(db);
```
Comment (betodealmeida) [inline superset-frontend/src/features/databases/DatabaseModal/DatabaseConnectionForm/CommonParameters.tsx:165]: Removed this.

File: superset-frontend/src/features/databases/DatabaseModal/index.tsx [hunk 2]
```diff
@@ -1656,6 +1666,16 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
             value: target.value,
           })
         }
+        onEncryptedExtraInputChange={({
+          target,
+        }: {
+          target: HTMLInputElement;
+        }) =>
+          onChange(ActionType.EncryptedExtraInputChange, {
```
Comment (rtexelm) [inline superset-frontend/src/features/databases/DatabaseModal/index.tsx:1674]: This is super helpful!

**Review by rtexelm — APPROVED** [review]

File: superset-frontend/src/features/databases/DatabaseModal/index.tsx (hunk 2 above)
Comment (betodealmeida) [reply superset-frontend/src/features/databases/DatabaseModal/index.tsx:1674]: Yeah, I was surprised it didn't exist yet!

**Pull request merged**